events/importer/harrastushaku.py

```diff
@@ -91,6 +92,56 @@ def import_places(self):
                 message = e if isinstance(e, HarrastushakuException) else traceback.format_exc()
                 logger.error('Error handling location {}: {}'.format(location.get('id'), message))
 
+    def map_harrastushaku_location_ids_to_tprek_ids(self, harrastushaku_locations):
+        '''
+        Example mapped dictionary result:
+        {
+            '95': 'harrastushaku:95',
+            '953': 'harrastushaku:953',
+            '968': 'tprek:20479',
+            '97': 'tprek:8062',
+            '972': 'tprek:9079',
+            '987': 'harrastushaku:987',
+            '99': 'tprek:8064',
+        }
+        '''
+        result = dict()
+
+        for harrastushaku_location in harrastushaku_locations:
+            harrastushaku_location_id = harrastushaku_location['id']
+
+            strict_filters = {
+                'id__startswith': self.tprek_data_source,
+                'name': harrastushaku_location['name'],
+                'address_locality': harrastushaku_location['city'],
+                'postal_code': harrastushaku_location['zip'],
+                'street_address': harrastushaku_location['address'],
+            }
+            flexible_filters = {
+                'id__startswith': self.tprek_data_source,
+                'address_locality': harrastushaku_location['city'],
+                'postal_code': harrastushaku_location['zip'],
+                'street_address': harrastushaku_location['address'],
+            }
+
+            strict_matched_places = Place.objects.filter(**strict_filters)
+
+            if len(strict_matched_places) >= 1:
+                harrastushaku_location_mapped_id = strict_matched_places.first().id
+
+            elif len(strict_matched_places) == 0:
+                flexible_matched_places = Place.objects.filter(**flexible_filters)
+
+                if len(flexible_matched_places) >= 1:
+                    harrastushaku_location_mapped_id = flexible_matched_places.first().id
+
+                elif len(flexible_matched_places) == 0:
+                    harrastushaku_location_mapped_id = f'{self.data_source.id}:{harrastushaku_location["id"]}'
+
+            result[harrastushaku_location_id] = harrastushaku_location_mapped_id
```

Comment:
The above `elif`s are unnecessary, if `len(something)` is not `>=1`, then it must be `0` :)

Actually the whole structure seems a bit overly complicated to me, how about just

```
tprek_place = Place.objects.filter(**strict_filters).first() or Place.objects.filter(**flexible_filters).first()

if tprek_place:
    result[harrastushaku_location_id] = tprek_place.id
else:
    result[harrastushaku_location_id] = f'{self.data_source.id}:{harrastushaku_location["id"]}'
```

Reply:
One other thing, using f-strings limits Python version to >= 3.6. Whether that is ok or not I don't know, required Python version for LE doesn't seem to be specified anywhere. Python 3.5 isn't *that* old, it is the default in Ubuntu 16.04, which still seems to be used quite a bit in some circles.

AFAIK other LE code runs at least on 3.5, so we probably shouldn't break that by using just a couple f-strings (I've been in a situation where exactly one f-string caused a lot of headache 🙂). Unless of course it will be decided that from now on LE requires at least Python 3.6. But this is something for @Rikuoja to decide.

Reply:
> The above elifs are unnecessary, if len(something) is not >=1, then it must be 0 :)

You are right, my bad. I was trying to be more explicit than implicit, but the way I was trying to do that was making everything else complex. Your suggested approach is much cleaner and easier to understand, and I replaced it with mine in 0fe669c6a508865c15af10b3c563109f9af203ba

> One other thing, using f-strings limits Python version to >= 3.6. Whether that is ok or not I don't know, required Python version for LE doesn't seem to be specified anywhere. Python 3.5 isn't that old, it is the default in Ubuntu 16.04, which still seems to be used quite a bit in some circles.

I thought Linked Events supports Python 3.6 because the `Dockerfile` image is using `python:3.6`, but again, my bad, as I knew that we don't do Docker deployments at the moment and therefore `Dockerfile` is a not reference to anything in production. I updated the code in cab321dbe67dbade40599d3d673c3caa4dc78210

Reply:
Yeah, 3.6 is our minimum. Mostly because we soon want to upgrade to newer django versions, which will drop 3.5 support anyway, and we don't run tests on 3.5 any more.

Reply:
Ok. That would be a good addition to the README :)